### PR TITLE
Change update to updateOne or updateMany

### PR DESCRIPTION
**Pull request by CuriousMagpie**
### Changes
`update` will be deprecated at some point in the future, changed instances of `update` to `updateOne` or `updateMany` as needed, mostly in `website/server/models/groups.js`.

----
UUID: f4e5c6da-0617-48bf-b3bd-9f97636774a8

### CODE REVIEW

File: migrations/command-line/apology_gems.js [hunk 1]
```diff
@@ -1 +1 @@
-db.users.update({ _id: { $in: [''] } }, { $inc: { balance: 0.5 } }, { multi: true });
+db.users.updateOne({ _id: { $in: [''] } }, { $inc: { balance: 0.5 } }, { multi: true });
```
Comment (SabreCat) [inline migrations/command-line/apology_gems.js]: not that this migration really needs to be updated at all (we can probably get rid of it tbh), but an `$in` query should be a many, not a one

File: migrations/command-line/contribs_plan.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // Give contrib.level 7+ free subscription for life
-db.users.update(
+db.users.updateOne(
```
Comment (SabreCat) [inline migrations/command-line/contribs_plan.js]: Many

File: migrations/command-line/habitica_day.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-db.users.update(
+db.users.updateOne(
```
Comment (SabreCat) [inline migrations/command-line/habitica_day.js]: many. And in fact the `multi` parameter shouldn't exist either.

File: website/server/models/group.js [hunk 4]
```diff
@@ -611,9 +611,9 @@ schema.methods.sendChat = function sendChat (options = {}) {
   };
 
   User
-    .update(query, lastSeenUpdateRemoveOld, { multi: true })
+    .updateMany(query, lastSeenUpdateRemoveOld, { multi: true })
```
Comment (SabreCat) [inline website/server/models/group.js]: `multi: true` is redundant with the `updateMany` function, can remove

File: website/server/models/group.js [hunk 5]
```diff
@@ -771,7 +771,7 @@ schema.methods.startQuest = async function startQuest (user) {
 
   // update the users who are not participating
   // Do not block updates
-  User.update({
+  User.updateMany({
     _id: { $in: nonMembers },
   }, _cleanQuestParty(),
   { multi: true }).exec();
```
Comment (SabreCat) [inline website/server/models/group.js]: 'nother `multi`. Might want to grep these!

File: website/server/models/group.js [hunk 6]
```diff
@@ -949,7 +949,7 @@ schema.methods.finishQuest = async function finishQuest (quest) {
   this.markModified('quest');
 
   if (this._id === TAVERN_ID) {
-    return User.update({}, updates, { multi: true }).exec();
+    return User.updateMany({}, updates, { multi: true }).exec();
```
Comment (SabreCat) [inline website/server/models/group.js]: multi

File: website/server/models/group.js [hunk 7]
```diff
@@ -1493,7 +1493,7 @@ schema.methods.updateTask = async function updateTask (taskToSync, options = {})
 
   // Updating instead of loading and saving for performances,
   // risks becoming a problem if we introduce more complexity in tasks
-  await taskSchema.update(updateQuery, updateCmd, { multi: true }).exec();
+  await taskSchema.updateMany(updateQuery, updateCmd, { multi: true }).exec();
```
Comment (SabreCat) [inline website/server/models/group.js]: multi

**Review by SabreCat — CHANGES_REQUESTED** [review]

File: migrations/command-line/apology_gems.js (hunk 1 above)
Comment (CuriousMagpie) [reply migrations/command-line/apology_gems.js]: done!

File: migrations/command-line/contribs_plan.js (hunk 2 above)
Comment (CuriousMagpie) [reply migrations/command-line/contribs_plan.js]: done!

File: migrations/command-line/habitica_day.js (hunk 3 above)
Comment (CuriousMagpie) [reply migrations/command-line/habitica_day.js]: done!

File: website/server/models/group.js (hunk 4 above)
Comment (CuriousMagpie) [reply website/server/models/group.js]: done!

File: website/server/models/group.js (hunk 5 above)
Comment (CuriousMagpie) [reply website/server/models/group.js]: done!

File: website/server/models/group.js (hunk 6 above)
Comment (CuriousMagpie) [reply website/server/models/group.js]: done!

File: website/server/models/group.js (hunk 7 above)
Comment (CuriousMagpie) [reply website/server/models/group.js]: done!

**Comment by SabreCat** [comment]
Hmm, I'm still seeing a bunch of `User.update` calls and `multi: true` options in the code--certainly we don't care about old migration archives, but it's everywhere in tests, and there's a few in the controllers e.g. lines 655-665. How have you been finding these to refactor?

**Comment by CuriousMagpie** [comment]
I grepped them, I'll do another pass and let you know if I'm having trouble finding them.

**Comment by SabreCat** [comment]
Superseded by https://github.com/HabitRPG/habitica/tree/phillip/mongoose

**Pull request closed without merging**